pydicom/tests/test_valuerep.py

```diff
@@ -56,68 +52,70 @@ class TestTM:
     """Unit tests for pickling TM"""
     def test_pickling(self):
         # Check that a pickled TM is read back properly
-        x = pydicom.valuerep.TM("212223")
-        assert time(21, 22, 23) == x
-        x.original_string = "hello"
-        assert "hello" == x.original_string
-        assert time(21, 22, 23) == x
-        data1_string = pickle.dumps(x)
-        x2 = pickle.loads(data1_string)
-        assert x == x2
-        assert x.original_string == x2.original_string
-        assert str(x) == str(x2)
-
-    def test_str(self):
-        """Test str(TM)."""
-        assert "212223.1234" == str(pydicom.valuerep.TM("212223.1234"))
-        assert "212223" == str(pydicom.valuerep.TM("212223"))
-        assert "212223" == str(pydicom.valuerep.TM("212223"))
-        assert "2122" == str(pydicom.valuerep.TM("2122"))
-        assert "21" == str(pydicom.valuerep.TM("21"))
-        assert "212223" == str(pydicom.valuerep.TM(time(21, 22, 23)))
-        assert "212223.000024" == str(
-            pydicom.valuerep.TM(time(21, 22, 23, 24)))
-        assert "010203" == str(pydicom.valuerep.TM(time(1, 2, 3)))
+        tm = pydicom.valuerep.TM("212223")
+        assert tm == time(21, 22, 23)
+        assert tm.original_string == "212223"
+        assert tm == time(21, 22, 23)
+        loaded_tm = pickle.loads(pickle.dumps(tm))
+        assert loaded_tm == tm
+        assert loaded_tm.original_string == tm.original_string
+        assert str(loaded_tm) == str(tm)
+
+    def test_pickling_tm_from_time(self):
+        tm = pydicom.valuerep.TM(time(21, 22, 23))
+        assert tm.original_string == "212223"
+        time_string = pickle.dumps(tm)
+        loaded_tm = pickle.loads(time_string)
+        assert loaded_tm == tm
+        assert loaded_tm.original_string == tm.original_string
+        assert str(loaded_tm) == str(tm)
+
+    def test_str_and_repr(self):
+        assert str(pydicom.valuerep.TM("212223.1234")) == "212223.1234"
```

Comment:
One thing I noticed (and I'm quite embarrassed by it): we do the asserts in the wrong order. If I recall correctly, it was myself that said it should be vice verse (or at least I was involved). Well, turns out I was completely wrong. I noticed because pytest showed the actual and expected values reversed when the tests failed...

Reply:
> If I recall correctly, it was myself who said it should be vice verse (or at least I was involved).

I remember that discussion, and I thought I had looked it up and I think I tested some too... maybe it is configurable or something?

Reply:
> maybe it is configurable or something?

Hm, I don't think so... Assertion rewrite is a central feature of pytest, and it is somewhat configurable via hooks, but not out of the box.
I also tested with older pytest versions, but they behave the same.